src/main/java/ua/tc/marketplace/controller/CategoryController.java

```diff
@@ -0,0 +1,58 @@
+package ua.tc.marketplace.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ua.tc.marketplace.model.dto.category.CreateCategoryDTO;
+import ua.tc.marketplace.service.CategoryService;
+
+/**
+ * CategoryController handles HTTP requests related to CRUD operations for the Category entity.
+ * It provides an API for retrieving, creating, updating, and deleting categories.
+ */
+@RestController
+@RequestMapping("/api/v1/category")
+@RequiredArgsConstructor
+public class CategoryController {
+    private static final Logger logger = LoggerFactory.getLogger(CategoryController.class);
+
+    private final CategoryService categoryService;
+
+    @GetMapping
+    public ResponseEntity<Page<CreateCategoryDTO>> getAllCategories(Pageable pageable) {
```

Comment:
This and other method should return just `CategoryDto` that include all fields of original category

Reply:
Why? Page<CategoryDTO> is a data page that can contain multiple CategoryDTO objects. This allows you to work with a large amount of data by breaking it into pages, which is useful for optimizing work with a large number of records.

Reply:
I mean `Page<CategoryDto>`